### PR TITLE
Add lookup functionality to UserQuery rest module, with a few basic tests

### PR DESCRIPTION
**Pull request by mrblippy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

Adds oEQ userquery/lookup REST endpoint support to the oeq-ts-rest-api module. 

This will enable us to lookup oEQ users, groups and roles by id. 

The primary driver for implementing this is to allow for the new Search page to read in a user id from a query string, and display the correct user details in the OwnerSelector control. 

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: oeq-ts-rest-api/src/UserQuery.ts [hunk 1]
```diff
@@ -61,6 +61,15 @@ export interface SearchParams {
   users: boolean;
 }
 
+export interface LookupParams {
+  /** String array of User IDs to lookup. */
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/UserQuery.ts:65]: No need to put the type info in the comment. So this (and the below three) can be:

```suggestion
  /** User IDs to lookup. */
```

Or if you want more, keep it abstract:

```suggestion
  /** List of User IDs to lookup. */
```

File: oeq-ts-rest-api/src/UserQuery.ts [hunk 2]
```diff
@@ -81,3 +90,18 @@ export const search = (
     isSearchResult,
     params
   );
+
+/**
+ * Lookup for users and related entities (i.e. groups and roles).
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/UserQuery.ts:95]: Be good to highlight in this description how this is different from `search()`.

File: oeq-ts-rest-api/src/AxiosInstance.ts [hunk 3]
```diff
@@ -68,4 +68,10 @@ export const PUT = <T, R>(path: string, data?: T): Promise<R> =>
     .then((response: AxiosResponse<R>) => response.data)
     .catch(catchHandler);
 
+export const POST = <T, R>(path: string, data?: T): Promise<R> =>
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/AxiosInstance.ts:71]: Please include support for a validator here (like in `GET`).

File: oeq-ts-rest-api/src/UserQuery.ts [hunk 4]
```diff
@@ -81,3 +90,18 @@ export const search = (
...
+ * @param params Query parameters to customize result
+ */
+export const lookup = (
+  apiBasePath: string,
+  params: LookupParams
+): Promise<SearchResult> =>
+  POST<LookupParams, SearchResult>(
+    apiBasePath + USERQUERY_ROOT_PATH + '/lookup',
+    params
+  );
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/UserQuery.ts:107]: Once you add support for a validator to `POST`, you want to add that in here too.

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Looks good @mrblippy 

But we will want to have a validator for `POST`, and tweak the comments a bit.

BTY.

File: oeq-ts-rest-api/src/AxiosInstance.ts (hunk 3 above)
Comment (mrblippy) [reply oeq-ts-rest-api/src/AxiosInstance.ts:71]: Ok, I will give it a go

**Review by edalex-ian — APPROVED** [review]
Looks good! Thanks @mrblippy :+1:

File: oeq-ts-rest-api/src/AxiosInstance.ts [hunk 5]
```diff
@@ -68,4 +68,29 @@ export const PUT = <T, R>(path: string, data?: T): Promise<R> =>
...
+ * @param data The data to be sent in POST request
+ */
+export const POST = <T, R>(
+  path: string,
+  validator: (data: unknown) => data is R,
+  data?: T
+): Promise<R> =>
+  axios
+    .post(path, data)
+    .then((response: AxiosResponse<R>) => {
```
Comment (ChristianMurphy) [inline oeq-ts-rest-api/src/AxiosInstance.ts:85]: Do we know that the `AxiosResponse` contains `R` yet?
I though that is what validator on line 87 is for?

**Review by ChristianMurphy — APPROVED** [review]

File: oeq-ts-rest-api/src/AxiosInstance.ts (hunk 5 above)
Comment (edalex-ian) [reply oeq-ts-rest-api/src/AxiosInstance.ts:85]: Good catch. :+1:

Comment (mrblippy) [reply oeq-ts-rest-api/src/AxiosInstance.ts:85]: Oh yes, i didn't update that after putting the validator in.  Thank you @ChristianMurphy

**Review by SammyIsConfused — APPROVED** [review]
Looks good Charlie!

**Pull request merged**